arangod/Aql/AqlItemBlockManager.cpp

```diff
@@ -185,35 +185,20 @@ size_t AqlItemBlockManager::Bucket::getId(size_t targetSize) noexcept {
   if (targetSize <= 1) {
     return 0;
   }
-  if (targetSize <= 10) {
-    return 1;
-  }
-  if (targetSize <= 20) {
-    return 2;
-  }
-  if (targetSize <= 40) {
-    return 3;
-  }
-  if (targetSize <= 100) {
-    return 4;
-  }
-  if (targetSize <= 200) {
-    return 5;
-  }
-  if (targetSize <= 400) {
-    return 6;
-  }
-  if (targetSize <= 1000) {
-    return 7;
-  }
-  if (targetSize <= 2000) {
-    return 8;
-  }
-  if (targetSize <= 4000) {
-    return 9;
-  }
-  if (targetSize <= 10000) {
-    return 10;
+
+  if (ADB_UNLIKELY(targetSize >= (1ULL << numBuckets))) {
+    return (numBuckets - 1);
   }
-  return 11;
+#if (defined(__GNUC__) || defined(__clang__))
+  size_t value = (8 * sizeof(unsigned long)) - static_cast<size_t>(__builtin_clzl(static_cast<unsigned long>(targetSize))) - 1;
+#elif defined(_MSC_VER)
+  unsigned long index;
+  _BitScanReverse64(&idx, static_cast<__int64>(targetSize));
+  size_t value = static_cast<size_t>(index);
+#else
+  static_assert(false, "no known way of computing clz");
+#endif
+
+  TRI_ASSERT(value < numBuckets);
+  return value;
```

Comment:
Can we extract this whole clz logic out into a method in lib/Basics somewhere and hide the compiler-specifics in the .cpp there? It's probably useful elsewhere in our code, and would be nice to re-use without reinventing the wheel every time. Actually, there's a good chance we already use something similar somewhere...

Reply:
Ok, just moved this into NumberUtils.cpp